libcontainer/user/lookup_deprecated.go

```diff
@@ -1,81 +0,0 @@
-package user
```

Comment:
This makes complete sense, but if the window between deprecation and removal is very small, we might slow down the adoption of new runc releases. I have mixed feelings with this, given that I don't think it will cause a huge pain to keep it for one more release, but it should be very simple to change for users too.

It seems there are more than 1.3k projects importing this: https://pkg.go.dev/github.com/opencontainers/runc/libcontainer/user?tab=importedby

I guess we are fine with this, but a little more elaboration on why it seems safe for the rest would be reassuring for me :)

Reply:
Well,
 * the number (1.3K) seems incorrect -- at the very least we should exclude those ancient repos with `Godeps` and multiple github forks of Docker etc. which are not distrinct pieces of software, but merely a way to work with Github;
 * github.com/moby/sys/user is a direct replacement, so the migration should be easy and straightforward.
 
 Having said that, we can definitely wait for runc v1.4 to remove this. @thaJeztah WDYT?

Reply:
Yeap, I haven't tried to see which of those 1.3k are not those forks.

I'm fine with both, given that the migration is so easy. Just wanted to make sure we were all on the same page regarding potential users and the small window. Given that is indeed very easy to migrate, I think this is just fine.

Reply:
OK, let's postpone that for v1.4.